### PR TITLE
Added support to use custom ca certificates

### PR DESCRIPTION
**Pull request by tusharraut1994**
Signed-off-by: Tushar Raut tushar@portworx.com

- Added `caCertsSecretName` flag into px-backup helm chart to set env and mount secret into containers (keycloak, px-backup, middleware and api-server)
- Updated README with steps and flag description

### CODE REVIEW

File: charts/px-backup/templates/px-backup/pxcentral-backup.yaml [hunk 1]
```diff
@@ -114,6 +114,10 @@ spec:
           value: {{ .Release.Namespace }}
         - name: PX_BACKUP_DEFAULT_ORG
           value: {{ .Values.pxbackup.orgName }}
+        {{- if .Values.caCertsSecretName }}
+        - name: SSL_CERT_DIR
+          value: /tmp/certs/*.crt
```
Comment (disrani-px) [inline charts/px-backup/templates/px-backup/pxcentral-backup.yaml:119]:
```suggestion
          value: /tmp/certs
```

File: charts/px-backup/templates/px-central-ui/pxcentral-lighthouse.yaml [hunk 2]
```diff
@@ -69,11 +69,21 @@ spec:
         env:
           - name: PX_NAMESPACE
             value: {{ .Release.Namespace }}
+          {{- if .Values.caCertsSecretName }}
+          - name: SSL_CERT_DIR
+            value: /tmp/certs/*.crt
```
Comment (disrani-px) [inline charts/px-backup/templates/px-central-ui/pxcentral-lighthouse.yaml:74]:
```suggestion
            value: /tmp/certs
```

File: charts/px-backup/templates/pxcentral-api-server.yaml [hunk 3]
```diff
@@ -132,6 +132,10 @@ spec:
         env:
           - name: PXC_NAMESPACE
             value: {{ .Release.Namespace }}
+          {{- if .Values.caCertsSecretName }}
+          - name: SSL_CERT_DIR
+            value: /tmp/certs/*.crt
```
Comment (disrani-px) [inline charts/px-backup/templates/pxcentral-api-server.yaml:137]:
```suggestion
            value: /tmp/certs
```

File: charts/px-backup/README.md [hunk 4]
```diff
@@ -72,6 +72,7 @@ Parameter | Description | Default
 `persistentStorage.storageClassName` | Provide storage class name which exists | `""`
 `storkRequired` | Scheduler name as stork | `false`
 `pxcentralDBPassword` | PX-Central cluster store mysql database password | `Password1`
+`caCertsSecretName` | Secret name where the CA certs are stored | `""`
```
Comment (sharma-tapas) [inline charts/px-backup/README.md:75]: Name of the Kubernetes Secret, which stores the CA Certificates.

**Review by sharma-tapas — APPROVED** [review]

**Pull request merged**